### PR TITLE
mzcompose: add kgen as a python service

### PR DESCRIPTION
**Pull request by ruchirK**
### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR adds a release note for any [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/user/content/release-notes.md#what-changes-require-a-release-note).

### CODE REVIEW

**Comment by ruchirK** [comment]
Hey @benesch throwing this up for a quick prelim review because I found myself guessing on a lot of things and wanted to make sure I wasn't too far off the mark.

First, reading the mzworkflows doc this seems like a thing worth doing (t/f)?

Then from there -- it seems like the subset of args I specified in the constructor are what we want kgen the service to have, and every other arg e.g. --num-records is something that the user needs to specify when the run kgen, similar to how testdrive doesn't take the list of files as an argument.

Finally, I wasn't sure about what to name the service / what to do with the `mzbuild` config option. 

Thanks!

**Comment by ruchirK** [comment]
I added a second commit showing an example workflow using the kgen service. When I try to run it with `./mzcompose run kafka-sources-load-test` inside `test/persistence` I get

```
ERROR: No such service: kgen
Failed! ERROR: services didn't come up cleanly: kgen, materialized
```

I'm not really sure how to debug this at this point unfortunately.

**Comment by benesch** [comment]
> I'm not really sure how to debug this at this point unfortunately. I feel like there's probably some dumb thing I'm doing wrong that you'll be able to spot immediately :D

Heh, yeah, you need to add `Kgen` to the magic `services` array. Just instantiating it in the file isn't enough!

File: misc/python/materialize/mzcompose.py [hunk 1]
```diff
@@ -1214,6 +1214,38 @@ def __init__(
...
+                "mzbuild": mzbuild,
+                "depends_on": depends_on,
+                "entrypoint": [
+                    "bash",
+                    "-O",
+                    "extglob",
+                    "-c",
+                    " ".join(entrypoint),
+                    "bash",
+                ],
```
Comment (benesch) [inline misc/python/materialize/mzcompose.py:1244]: This is weird testdrive crap that you should ditch! Just the entrypoint as you've defined it on L1224 is great.

File: test/persistence/mzworkflows.py [hunk 2]
```diff
@@ -149,3 +153,19 @@ def workflow_disable_user_indexes(w: Workflow):
...
+    w.wait_for_mz(service="materialized")
+
+    w.run_service(
+        service="testdrive-svc",
+        command="kafka-sources/kafka-sources-load-test.td",
+    )
+
+    w.run_service(
+        service="kgen",
+        command="--num-records=100 --keys=random --keys-min=1 --keys-max=10000000 --values=bytes --max-message-size=501 --min-message-size=500 --num-records=500 --topic=testdrive-load-test-1",
```
Comment (benesch) [inline test/persistence/mzworkflows.py:170]: nit: passing these as an array of strings is preferred

**Review by benesch — COMMENTED** [review]
This looks great, modulo the stuff I've commented on. This is def something we want in mzcompose!

**Comment by ruchirK** [comment]
@benesch made the requested edits. I'll use a separate pr for the actual benchmark using kgen but wanted to confirm: does this look good to merge now?

**Review by benesch — APPROVED** [review]
Yeah, LGTM!

**Pull request merged**